src/string_pool.cr

```diff
@@ -1,19 +1,52 @@
+# A String pool is a collection of Strings
+# It can be used to return an already present String instead of creating a new String.
+#
+#
+#   require "string_pool"
+#   stringpool = StringPool.new
+
 class StringPool
+  # Returns the size
+  #
+  #     stringpool = StringPool.new
+  #     puts stringpool.size # => 0
   getter size
 
+  # Initialization of the elements 
+  # At the begining, the size is null and the bucket is an array of size 11
   def initialize
     @buckets = Array(Array(String)?).new(11, nil)
     @size = 0
   end
 
+  # Returns true if the String Pool have no element else returns false
+  #
+  #     stringpool = StringPool.new
+  #     stringpool.empty? # => true
+  #     stringpool.get("crystal") 
+  #     stringpool.empty? # => false
   def empty?
     @size == 0
   end
 
+  # Calls get(str : UInt8*, len) with the specified slice. 
```

Comment:
Maybe explain "Returns a string with the contents of the given slice. If a string with those contents was already present in the pool, that one is returned. Otherwise a new string is created, put in the pool and returned". I would avoid explaining methods with "ends up invoking X".